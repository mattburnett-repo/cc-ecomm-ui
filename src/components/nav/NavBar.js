export default function NavBar (props) {
    const { calledFrom } = props;

    return (
        <div>
            <div role="presentation" aria-label="header-message">header message</div>
            if(calledFrom !== 'landing') {
                <>
                    <button aria-label="home">Home</button>
                    <button aria-label="show-cart">Show Cart</button>
                </>
            };
            if(calledFrom === 'cart') {
                <button aria-label="checkout">Checkout</button>
            }
            if(calledFrom === 'shippingInfo' || calledFrom === 'paymentInfo') {
                <button aria-label="cancel">Cancel</button>
            }
            <button aria-label="logout">Log Out</button>
        </div>
    )
}