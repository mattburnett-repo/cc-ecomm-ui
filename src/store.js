import { configureStore } from '@reduxjs/toolkit';

import authReducer from './features/auth/authSlice';
import productCategoryReducer from './features/productCategory/productCategorySlice'
import productsListingReducer from './features/products/productsSlice'
import ordersReducer from './features/orders/ordersSlice'
import cartsReducer from './features/carts/cartsSlice'

function rootReducer() {
    return {
        reducer: {
            auth: authReducer,
            productCategories: productCategoryReducer,
            orders: ordersReducer,
            carts: cartsReducer,
            products: productsListingReducer,
        }    
    }
}

const store = configureStore(rootReducer());

export default store;