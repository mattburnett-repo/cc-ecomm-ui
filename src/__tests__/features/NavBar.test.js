// 20211217: any wireframe will do...

// import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { Provider } from 'react-redux'
import store from '../../store'

import NavBar from '../../features/nav/navBar';

// TODO: needs to test rendering based on calledFrom prop...
describe('<NavBar /> feature tests', () => {
    it('renders a snapshot', () => {
        const tree = renderer.create(<Provider store={store}><NavBar /></ Provider>).toJSON();
        expect(tree).toMatchSnapshot();
    });
}); // end snapshot

