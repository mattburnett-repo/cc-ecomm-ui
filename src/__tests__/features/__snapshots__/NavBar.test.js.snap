// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> feature tests renders a snapshot 1`] = `
<div
  aria-label="nav-bar"
  role="presentation"
>
  <div
    aria-label="header-message"
    role="presentation"
  >
     
    
  </div>
  <div
    aria-label="header-current-cart-item-count"
    role="presentation"
  >
    <div>
      You have 
      0
       items in your current cart
    </div>
  </div>
  <button
    aria-label="home"
    onClick={[Function]}
  >
    Home
  </button>
  <button
    aria-label="show-cart"
    onClick={[Function]}
  >
    Show Cart
  </button>
  <button
    aria-label="logout"
    onClick={[Function]}
  >
    Log Out
  </button>
</div>
`;
